lib/ravelin/ravelin_object.rb

```diff
@@ -0,0 +1,50 @@
+module Ravelin
+  class RavelinObject
+    class << self
+      attr_reader :attributes
+
+      def required_attributes
+        @required_attributes ||= []
+      end
+
+      def attr_required(*names)
+        @required_attributes = *names
+      end
+
+      def attr_accessor(*names)
```

Comment:
The 3 things that come to mind are:
1) as it is
2) using metaprogramming to look for instance variables
3) using structs and struct#members

To be honest, I don't think your solution is bad. 

The only thing that might be a problem on the future is that you don't have any definition of whitelisting attributes and you are just dumping everything. If one day there is a private attribute added for some reason, it will be output to the `serialised_hash`. Maybe that scenario will never happen.